cmd/signozschemamigrator/schema_migrator/index_operations.go

```diff
@@ -262,3 +279,67 @@ func (a AlterTableClearIndex) ToSQL() string {
 	}
 	return sql.String()
 }
+
+func JSONSubColumnIndexName(column, path, typeColumn string, index IndexType) string {
+	expr := column + "." + path
+	return fmt.Sprintf("`%s_%s_%s`", expr, typeColumn, index)
+}
+
+func jsonSubColumnIndexExprFormat(expr, typeColumn string) string {
+	parts := strings.Split(expr, ".")
+	for idx, part := range parts {
+		if keycheck.IsBacktickRequired(part) {
+			part := strings.Trim(part, "`") // trim if already present
+			parts[idx] = "`" + part + "`"
+		}
+	}
+
+	indexExpr := stringBasedIndexExpr
+	if typeColumn != "String" {
+		indexExpr = numberBasedIndexExpr
+	}
+
+	return fmt.Sprintf(indexExpr, strings.Join(parts, "."), typeColumn)
+}
+
+func JSONSubColumnIndexExpr(column, path, typeColumn string) string {
+	expr := column + "." + path
+	return jsonSubColumnIndexExprFormat(expr, typeColumn)
+}
+
+// Returns the subcolumn name from the index expression
+// If the expression is not a JSON subcolumn index expression, returns an error
+func UnfoldJSONSubColumnIndexExpr(expr string) (string, string, error) {
```

Comment:
can we use a grammar for this, should be a simple one and will be easier to maintain